docs/DeveloperGuide.md

```diff
@@ -251,43 +251,122 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+| `* * `  | Hall admin managing hall events     | Quickly filter out students involved in certain events   | I can email selected group of students easily                |
+| `* * *`  | Hall admin managing hall events     | Quickly find out who attended compulsory events   | I can email selected group of students easily                |
+| `* * *`  | Hall admin managing hall events     | Create hall events  | I can contact the residents about the events                |
+| `* * *`  | Hall admin managing hall events     | Update hall events   | I can update the event's details                |
+| `* *`  | Hall admin managing hall events     | Archive hall events   | I can archive past events                |
+| `* * `  | Hall admin managing hall events     | Delete hall events   | I can remove events that were cancelled                |
+| `* * `  | Hall admin managing hall events     | Query past hall events based on period   | So that I can do a recap of hall events                |
+| `* * `  | Hall admin managing discipline      | Query for a student's demerit points and all rules broken   | I can expel them next sem                |
+| `*`  | Hall admin managing discipline      | Query most commonly broken rules   | I can work to prevent those cases                |
+| `* `  | Hall admin managing discipline      | Create rule lists for demerit points   | To create and include new rules such as covid temp measures                |
+| `* * `  | Hall admin managing discipline      | Award demerit points to a student records   | Punish students                |
+| `* `  | Hall admin managing discipline      | Select a rule that was broken when awarding demerit points   | To correctly allocate the demerit point for the student commiting the offence                |
+| `*  `  | Hall admin managing discipline      | Give additional details, e.g. who, what, when, how, where   | I know why they received the demerit points in the first place                |
+| `* * `  | Hall admin managing discipline      | Delete demerit points from students' records   | Undo any mistakes that occurred when giving them demerit points               |
+| `* `  | Hall admin managing discipline      | Reset demerit points for all students   | -                |
+| `* `  | Hall admin managing discipline      | Keep track of the budget left for the block events   | I can plan the event according to the budget                |
+| `* * `  | Hall admin general                  | Export csv, based on filters  | I can send this data to people who will want information on these residents                |
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `hall leader`, unless specified otherwise)
```

Comment:
Should this be Hall-y instead of AddressBook? Also, should it be hall admin (according to the use cases) instead of hall leader?